python/lib/mesh.cc

```diff
@@ -119,8 +119,19 @@ WrapMesh(py::module& mesh)
   );
   mesh_continuum.def(
     "ComputeVolumePerBlockID",
-    &MeshContinuum::ComputeVolumePerBlockID,
-    "Compute volume per block ID."
+    [](MeshContinuum& self) {
+      std::map<int, double> cpp_map = self.ComputeVolumePerBlockID();
+      py::dict py_map;
+      for (const auto& [block_id, volume] : cpp_map)
+      {
+        py_map[py::cast(block_id)] = volume;
+      }
+      return py_map;
+    },
+    R"(
+    Compute volume per block ID and return a Python dict mapping
+    each block_id (int) to its total volume (float).
+    )"
```

Comment:
What is the reason for not using `#include <pybind11/stl.h>`? 

Doco says if you use it, then conversion between STL containers and python types are done automatically [see here](https://pybind11.readthedocs.io/en/stable/advanced/cast/stl.html#automatic-conversion), i.e. in this case `std::map` would be automatically converted into `dict`. Then, you would not need the lambda and you would just bind the member function...

Reply:
I suppose then the answer is simply (aftyer including `<pybind11/stl.h>`):

```
    .def("ComputeVolumePerBlockID",
         &MeshContinuum::ComputeVolumePerBlockID,
         R"(
           Compute volume per block ID and return a std::map<int,double>,
           which will automatically be converted to a Python dict.
         )");

```

@quocdang1998 can you comment on this and on the use of `<pybind11/stl.h>`?

Reply:
Note that the `R"(...)"` is the doco for python, so `std::map` in that context makes no sense. You probably want to say something like:
```
Compute volume per block ID. Returns dict where key is the block ID and 
the value is the computed volume.
```

Or you can take it one step forward and use proper python doco:
```
Compute volume per block ID

Returns
-------
dict
        Key is the block ID and the value is the computed volume
```

Reply:
I have tried this and it works. I have left the older version in comments for now. I will fix this later, after I get feedback from @quocdang1998 . it looks a lot easier with pybind11/stl.h

Reply:
is this good? [should dict be alone on the line?]
```
  mesh_continuum.def(
    "ComputeVolumePerBlockID",
    &MeshContinuum::ComputeVolumePerBlockID,
    R"(
    Compute volume per block ID

    Returns
    -------
    dict
            Key is the block ID and the value is the computed volume
    )"
  );
```

Reply:
I think it should. The code is using the numpy style, see [code sample](https://github.com/Open-Sn/opensn/blob/acb16041898deccf0795c4bbda6f20bbeec7870e/python/lib/post.cc#L55-L58).

See [here](https://www.datacamp.com/tutorial/docstrings-python) for different styles. I am not sure why they do not provide types for return values. Maybe, you want to change `dict` to `volumes`, though. IDK.